### PR TITLE
[Handshake] Removed use count check for handshake dialect

### PR DESCRIPTION
**Pull request by JianyiCheng**
The Handshake Dialect required exactly one use for each op. This patch removes this condition in verification.

### CODE REVIEW

**Comment by hanchenye** [comment]
Why this verification is not the case?

**Comment by lattner** [comment]
Wow, I was not aware of that.  While that looks convenient for code generation, that seems like it would make compiler transformations very difficult / constrained.  I'm not a handshake person though, so I think others should review.

**Comment by teqdruid** [comment]
'required' like past tense and it doesn't anymore? I assume it was because handshake lowerings didn't support handshaked fanouts. These can be tricky to implement correctly since you've got two backpressure sources to reason about. Depending on the signaling standard, it might require separate buffers for each. In a past life, I had this requirement with the exception for a 'fork' op which handled that complexity.

If this is the case and handshake doesn't support fan-out, I would move this check to the lowering since it doesn't really come into play until you have to produce working FIRRTL or RTL.

**Comment by stephenneuendorffer** [comment]
@teqdruid Yes, this is basically something that should be part of backend lowerings, not the core verifier.  There are a few places that make this assumption at the moment, primarily the handshake-runner.  The point of this patch is to fix that, but it probably shouldn't land until the users are fixed first.

**Comment by teqdruid** [comment]
Yep #393 answered my question.

**Comment by teqdruid** [comment]
@stephenneuendorffer Seems like you might need this lifted in order for the other PRs' tests to pass.

**Review by teqdruid — APPROVED** [review]

**Pull request merged**
